clang/tools/scan-build-py/lib/libear/ear.c

```diff
@@ -1759,156 +1744,261 @@ static void bear_report_call(char const *fun, char const *const argv[]) {
 
   emit_cmake_warning(argv, argc);
 
-  // compiler list should be intercepted.
-  const char *const compiler_array[] = {"nvcc", "clang++"};
-  // Current compiler index intercepted.
-  int compiler_idx = 0;
-
-  // To indicate whether the captured argv[i] is a compiler or ld command,
-  // value: 1 yes, value 0 no.
-  int is_nvcc_or_ld = 0;
-
-  // To indicate whether the object file has been fake generated,
-  // value: 1 obj file generated, value: 0 not generated.
-  int flag_object = 0;
-
-  // flag_optval is use for case: for options "-o xxx.o", "-o" and "xxx.o" is in
-  // argv[i] and argv[i+1], if "-o" is found in argv[i], then flag_optval
-  // is set to show argv[i+1] contains the xxx.o
-  int flag_optval = 0;
-
-  // value 1: means current command line is a compiler command, and the fake obj
-  // file has been created, else ret is set to 0.
-  int ret = 0;
-
-  // To indicate whether the captured cmd is an ar static library command.
-  int is_ar_staticlib_cmd = 0;
+  int is_tool_existence = 0;
+  if (file_exists(argv[0])) {
+    is_tool_existence = 1;
+  }
 
-  const char *command_cp = NULL;
-  size_t it_cp = 0;
-  // (CPATH=;command  args), need remove () around the command
-  int has_parenthesis = 0;
+  if (is_tool_existence) {
+    for (size_t it = 0; it < argc; ++it) {
+      fprintf(fd, "%s%c", argv[it], US);
```

Comment:
debug purpose printf?

Reply:
No, it used to collect options for current command.